### DISCUSSION TITLE
Week 6: Accessibility

### DISCUSSION THREAD

**Discussion by coopsand** [USU-ITLS-x265-23-summer/class-repo#126] [General]
## Thoughts on Reading and Video
Accessibility has always been something that I strive to uphold when I design different websites and prototypes. I want to work as a UX designer, so I try to educate myself on how to build designs that are accessible. The video that we watched had a very impactful affect on me because it showed the struggles that people with disabilities can face in a very unique way. It flipped the switch on who society was built around. 

In design, it's hard to design things to encompass 100% of your possible audience, but there are certainly steps that you can take to improving your designs and making them more accessible to everyone. Having easy-to-read color schemes and text goes a long way in making the website accessible to everyone. 

I liked the reading as well because it showcased that someone's disability doesn't define who they are. A lot of times, we only see labels. We forget that there are people attached to said labels. Just because someone happens to have a disability, doesn't mean that they can't do certain things. Always be open to letting everyone try new things and bring feedback to the table. This will have a massive effect.

## Thoughts on FreeCodeCamp
I felt that the markers activity this week was a little all over the place. I felt like it taught a lot of useful information, but it taught it fast. I wish that the current marker lesson was split into a few different lessons. One for each way to change an items color (RGB, Hex, etc.) and then a couple smaller lessons on opacity, gradients, etc. Having the lesson be split up into smaller bite sized chunks would also make it easier to review concepts.

## Project Updates
One thing I just realized this week is that I have absolutely no clue where I need to program my site, how to save it, post it, etc. Does anyone have any recommendations or ideas? Will there be an assigned place to do this? I have used a couple of compilers and I think I remember Visual Studio having an HTML compiler, but I'm not 100% sure. 

With the portfolio site, I don't know how I will add images. We talked about it in FreeCodeCamp, but I guess I don't understand if I will need to upload the images somewhere prior to placing them in the site. Any thoughts?

With the book blog, I am not sure how I will want to organize it. That's just something I will have to consider as I continue to think ahead about my final project.

**Comment by derekscholes** [comment]
The program I have been using is Light Table. Its an opensource one and it allows you to open a browser in it and refresh the browser live to see the updates. You can also see your javascript or CSS and have many tabs open at once. 
<img width="1680" alt="Screenshot 2023-06-14 at 11 05 01 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/132800749/94b7095a-5e79-40cf-8349-1c35214e8eb5">

So with everything, you can look at youtube videos that show you how to do so. What I have done so far is create a folder specifically for the site and all the files such as additional html and css can be linked to your site just with their name such as < a href="style.css"> or you can put them in a subfolder and just had the folder name before the file name e.g. < a href="css-folder/style.css">
<img width="1032" alt="Screenshot 2023-06-14 at 11 10 25 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/132800749/239870b2-596d-4030-85e5-b4a8f3851786">

You can also use GitHub to publish your site to the internet for free which is kind of cool. All you do is create a new repository and upload all your completed files to it how the are in your folder and set certain things up and it will work. 
https://www.youtube.com/watch?v=p1QU3kLFPdg

**Reply by metasean** [reply]
This is largely how projects will be done.  The big difference is that I'd like everyone to use [GitHub CodeSpaces](https://github.com/features/codespaces). It is configured so that it's connected to your project's GitHub repo.  Because of this, I can quickly put your code into a viewable container.    

If you're an eager beaver, start with the ["project" template](https://github.com/USU-ITLS-x265-23-summer/project).  I recommend the "Live Server" extension.  Other than that, if you're an eager beaver, I'll leave you to figure out the rest.  More complete instructions will be coming shortly.  Make sure you post a link to your project in the ["Project Repo Links"](https://github.com/orgs/USU-ITLS-x265-23-summer/discussions/132) discussion post.

NOTE 1: If you're an eager beaver, make sure that you are committing your code to the repo.  The CodeSpace instance allows you to work completely independently on the code, but you still have to follow all the git steps. 

NOTE 2: Also, if you're an eager beaver, you do NOT need to, and almost always do NOT want to, CREATE a new CodeSpace for your project.

**Comment by jakegros** [comment]
Here are some GitHub page examples that I found while exploring and wondering the same thing:
[Collection: GitHub Pages Examples](https://github.com/collections/github-pages-examples)
